CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/SyntaxNodeFile.swift

```diff
@@ -0,0 +1,494 @@
+                      keyExpression: ExprSyntax(#""\#(raw: child.swiftName)""#),
+                      valueExpression: child.isOptional
+                        ? ExprSyntax("\(raw: child.swiftName).map(Syntax.init)?.asProtocol(SyntaxProtocol.self) as Any")
+                        : ExprSyntax("Syntax(\(raw: child.swiftName)).asProtocol(SyntaxProtocol.self)")
+                    )
+                  }
+                }
+              )
+            }
+          )
+        }
+      }
+    }
+  }
+}
+
+private func generateSyntaxChildChoices(for child: Child) throws -> EnumDeclSyntax? {
+  guard case .nodeChoices(let choices) = child.kind else {
+    return nil
+  }
+
+  return try EnumDeclSyntax("public enum \(raw: child.name): SyntaxChildChoices") {
+    for choice in choices {
+      DeclSyntax("case `\(raw: choice.swiftName)`(\(raw: choice.typeName))")
+    }
+
+    try! VariableDeclSyntax("public var _syntaxNode: Syntax") {
+      try! SwitchStmtSyntax("switch self") {
+        for choice in choices {
+          SwitchCaseSyntax("case .\(raw: choice.swiftName)(let node):") {
+            StmtSyntax("return node._syntaxNode")
+          }
+        }
+      }
+    }
+
+    DeclSyntax("init(_ data: SyntaxData) { self.init(Syntax(data))! }")
+
+    for choice in choices {
+      if let choiceNode = SYNTAX_NODE_MAP[choice.syntaxKind], choiceNode.isBase {
+        DeclSyntax(
+          """
+          public init<Node: \(raw: choiceNode.name)Protocol>(_ node: Node) {
+            self = .\(raw: choice.swiftName)(\(raw: choiceNode.name)(node))
+          }
+          """
+        )
+
+      } else {
+        DeclSyntax(
+          """
+          public init(_ node: \(raw: choice.typeName)) {
+            self = .\(raw: choice.swiftName)(node)
+          }
+          """
+        )
+      }
+    }
+
+    try! InitializerDeclSyntax("public init?<S: SyntaxProtocol>(_ node: S)") {
+      for choice in choices {
+        StmtSyntax(
+          """
+          if let node = node.as(\(raw: choice.typeName).self) {
+            self = .\(raw: choice.swiftName)(node)
+            return
+          }
+          """
+        )
+      }
+
+      StmtSyntax("return nil")
+    }
+
+    try! VariableDeclSyntax("public static var structure: SyntaxNodeStructure") {
+      let choices = ArrayExprSyntax {
+        for choice in choices {
+          ArrayElementSyntax(
+            expression: ExprSyntax(".node(\(raw: choice.typeName).self)")
+          )
+        }
+      }
+
+      StmtSyntax("return .choices(\(choices))")
+    }
+  }
+}
+
+fileprivate extension Node {
+  func generateInitializerDeclHeader(optionalBaseAsMissing: Bool) -> PartialSyntaxNodeString {
+    if children.isEmpty {
+      return "public init()"
+    }
+
+    var genericParamNames: [String: Int] = [:]
+    var genericParams: [String] = []
+
+    func createFunctionParameterSyntax(for child: Child, optionalBaseAsMissing: Bool) -> FunctionParameterSyntax {
```

Comment:
It's not possible to store paramType inside a result builder. 
So made this function here to have the logic of creating a function parameter syntax